src/main/plugin/iso19139.ca.HNAP/process/thumbnail-add.xsl

```diff
@@ -90,11 +102,19 @@
   </xsl:template>
 
 
-  <xsl:template match="gmd:graphicOverview[concat(
-                         */gmd:fileName/gco:CharacterString,
-                         */gmd:fileName/gmd:PT_FreeText/gmd:textGroup/gmd:LocalisedCharacterString[@locale = '#DE'],
-                         */gmd:fileDescription/gmd:PT_FreeText/gmd:textGroup/gmd:LocalisedCharacterString[@locale = '#DE'],
-                         */gmd:fileDescription/gco:CharacterString) = normalize-space($updateKey)]">
+  <!-- Updating the gmd:graphicOverview based on update parameters -->
+  <!-- Note: first part of the match needs to match the xsl:for-each select from extract-relations.xsl in order to get the position() to match -->
+  <!-- The unique identifier is marked with resourceIdx which is the position index and resourceHash which is hash code of the current node (combination of url, resource name, and description) -->
+  <xsl:template
+    priority="2"
+    match="*//gmd:graphicOverview
+         [$resourceIdx = '' or position() = xs:integer($resourceIdx)]
+         [    ($resourceHash != '' or ($updateKey != '' and normalize-space($updateKey) = concat(
+                           */gmd:fileName/gco:CharacterString,
+                           */gmd:fileName/gmd:PT_FreeText/gmd:textGroup/gmd:LocalisedCharacterString[@locale = '#DE'],
+                           */gmd:fileDescription/gmd:PT_FreeText/gmd:textGroup/gmd:LocalisedCharacterString[@locale = '#DE'],
```

Comment:
Is this copied from iso19139? I think we can remove this

Reply:
yes, looks like it. I don't see any commit about the code, so guess it was long time ago's commit